tests/testCoaddInputs.py

```diff
@@ -0,0 +1,256 @@
+        self.coadd.getInfo().setCoaddInputs(coaddInputs)
+
+
+class CoaddInputsTestCase(lsst.utils.tests.TestCase):
+    def setUp(self):
+        self.version = 2  # desired version of ExposureTable/CoaddInput
+        self.numExp = 3
+        mockCoadd = MockCoadd(numExp=self.numExp, version=self.version)
+        self.coadd = mockCoadd.coadd
+        self.exposures = mockCoadd.exposures
+        self.dataDir = os.path.join(os.path.dirname(__file__), "data")
+
+    def tearDown(self):
+        del self.coadd
+        del self.exposures
+
+    def assertPsfsNearlyEqual(self, psf1, psf2):
+        im1 = psf1.computeImage()
+        im2 = psf2.computeImage()
+        self.assertImagesNearlyEqual(im1, im2)
+
+    def getCoaddPath(self, version):
+        return os.path.join(self.dataDir,
+                            "testCoaddInputs_coadd_with_version_{}_data.fits".format(version))
+
+    def testPersistence(self):
+        """Read and write a coadd and check the CoaddInputs"""
+        coaddPath = self.getCoaddPath(version=self.version)
+        self.coadd.writeFits(coaddPath)
+        coadd = lsst.afw.image.ExposureF(coaddPath)
+        coaddInputs = coadd.getInfo().getCoaddInputs()
+        self.assertCoaddInputsOk(coaddInputs, version=self.version)
+        if not SaveCoadd:
+            os.unlink(coaddPath)
+        else:
+            print("SaveCoadd true; saved coadd as: %r" % (coaddPath,))
+
+    def xtestReadV1Coadd(self):
+        """Read a coadd that contains version 1 CoaddInputs"""
+        coaddPath = self.getCoaddPath(version=1)
+        coadd = lsst.afw.image.ExposureF(coaddPath)
+        coaddInputs = coadd.getInfo().getCoaddInputs()
+        self.assertCoaddInputsOk(coaddInputs, version=1)
+
+    def assertCoaddInputsOk(self, coaddInputs, version):
+        self.assertIsNotNone(coaddInputs)
+        visitTable = coaddInputs.visits
+        self.assertEqual(len(visitTable), 3)
+        for i, visit in enumerate(visitTable):
+            exp = self.exposures[i]
+            expInfo = exp.getInfo()
+            self.assertEqual(visit.getId(), i)
+            self.assertEqual(visit.getBBox(), exp.getBBox())
+            self.assertWcsNearlyEqualOverBBox(visit.getWcs(), expInfo.getWcs(), visit.getBBox())
+            self.assertPsfsNearlyEqual(visit.getPsf(), exp.getPsf())
+            self.assertPairsNearlyEqual(visit.getCalib().getFluxMag0(), (1.1e12, 2.2e10))
+            self.assertEqual(len(visit.getApCorrMap()), 3)
+            self.assertEqual(set(visit.getApCorrMap().keys()), set(expInfo.getApCorrMap().keys()))
+            self.assertFloatsAlmostEqual(np.array(visit.getValidPolygon().getVertices()),
+                                         np.array(expInfo.getValidPolygon().getVertices()))
+            if version > 1:
+                self.assertEqual(visit.getVisitInfo(), expInfo.getVisitInfo())
+            else:
+                self.assertFalse(visit.getVisitInfo().getDate().isValid())
```

Comment:
Just a reminder that if you do change the code in afw to make this an empty pointer, you'll want to change this line too (to check that `getVisitInfo() is None`).